core/src/main/java/com/datastax/oss/driver/internal/core/pool/ChannelPool.java

```diff
@@ -259,7 +259,8 @@ private SingleThreaded(
               () -> reconnectionPolicy.newNodeSchedule(node),
               this::addMissingChannels,
               () -> eventBus.fire(ChannelEvent.reconnectionStarted(node)),
-              () -> eventBus.fire(ChannelEvent.reconnectionStopped(node)));
+              () -> eventBus.fire(ChannelEvent.reconnectionStopped(node)),
+              error -> {});
       this.configListenerKey =
```

Comment:
This may need to be changed.... I'm not 100% clear on what needs to happen here to successfuly close the channel.

Reply:
I think the right thing to do is to force the node down. If the policy decides to abort it means we have really given up on this node. You can see in `ChannelPool.onAllConnected()` how it already does that for invalid keyspace, instead of doing it there we should rethrow it to the policy (see my other comment).

When `onAllConnected` throws, there might be multiple errors, so I think we should use a wrapper:
```java
class PoolReconnectionException extends DriverException {
  /** The individual error for each connection that was attempted. */
  List<Throwable> getConnectionErrors();

  /** If the reconnection attempted to complete a pool that already had active channels, their count. */
  int getActiveChannels();
}
```